### PR TITLE
Feature/60/add involvement check before updating delating offers

### LINKED ISSUE #60: (SP: _) Add involvement check before updating/deleting offers
## Description
Add involvment check so that if a user not involved in an offer (not an author) they can't update or delete a doucment. If they try, a forbidden error should be thrown
## Acceptance Criteria
- [ ] non-involved user must not be able to update/delete the offer
- [ ] this logic must be covered by tests

### PR DESCRIPTION
**Pull request by OliaHrytsak**
Add involvment check so that if a user not involved in an offer (not an author) they can't update or delete a doucment. If they try, a forbidden error should be thrown

Now, I'm login as a student and this is my offer
image

I want update prise, for example
image

Result
image

That is not my offer( - it is from tutor
image

I'm trying to update it

image

I'm trying to delete it
image

### CODE REVIEW

**Review by iva-stasia — APPROVED** [review]

**Review by StasRusnak1 — APPROVED** [review]

**Review by VictoriaRodik — APPROVED** [review]

**Review by Wiret02 — APPROVED** [review]

File: services/offer.js [hunk 1]
```diff
@@ -52,7 +53,13 @@ const offerService = {
   updateOffer: async (id, currentUserId, updateData) => {
     const filteredUpdateData = filterAllowedFields(updateData, allowedOfferFieldsForUpdate)
 
-    const offer = await Offer.findById(id)
+    const offer = await Offer.findById(id).exec()
+    console.log('Current User ID:', currentUserId)
```
Comment (holus-bolus) [inline services/offer.js]: Would you mind considering delete console logs?

File: services/offer.js [hunk 2]
```diff
@@ -52,7 +53,13 @@ const offerService = {
...
+    console.log('Offer Author ID:', offer.author.toString())
```
Comment (Wiret02) [inline services/offer.js]: Just remove console.log

File: services/offer.js (hunk 1 above)
Comment (OliaHrytsak) [reply services/offer.js]: Done! Thank you for your attention

File: services/offer.js (hunk 2 above)
Comment (OliaHrytsak) [reply services/offer.js]: Done! Thank you for your attention

**Review by vlad-khrychov — APPROVED** [review]

**Pull request merged**